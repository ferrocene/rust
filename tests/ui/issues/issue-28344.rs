//@ revisions: edition2015 edition2021
//@[edition2015] edition: 2015
//@[edition2021] edition: 2021
use std::ops::BitXor;

fn main() {
    let x: u8 = BitXor::bitor(0 as u8, 0 as u8);
    //[edition2015]~^ ERROR must be specified
    //[edition2015]~| WARN trait objects without an explicit `dyn` are deprecated
    //[edition2015]~| WARN this is accepted in the current edition
    //[edition2021]~^^^^ ERROR expected a type, found a trait

    let g = BitXor::bitor;
    //[edition2015]~^ ERROR must be specified
    //[edition2015]~| WARN trait objects without an explicit `dyn` are deprecated
    //[edition2015]~| WARN this is accepted in the current edition
    //[edition2021]~^^^^ ERROR expected a type, found a trait
}
