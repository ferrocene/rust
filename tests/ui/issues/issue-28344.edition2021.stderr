error[E0782]: expected a type, found a trait
  --> $DIR/issue-28344.rs:7:17
   |
LL |     let x: u8 = BitXor::bitor(0 as u8, 0 as u8);
   |                 ^^^^^^
   |
help: you can add the `dyn` keyword if you want a trait object
   |
LL |     let x: u8 = <dyn BitXor>::bitor(0 as u8, 0 as u8);
   |                 ++++       +
help: you may have misspelled this associated item, causing `BitXor` to be interpreted as a type rather than a trait
   |
LL |     let x: u8 = BitXor::bitxor(0 as u8, 0 as u8);
   |                            +

error[E0782]: expected a type, found a trait
  --> $DIR/issue-28344.rs:13:13
   |
LL |     let g = BitXor::bitor;
   |             ^^^^^^
   |
help: you can add the `dyn` keyword if you want a trait object
   |
LL |     let g = <dyn BitXor>::bitor;
   |             ++++       +
help: you may have misspelled this associated item, causing `BitXor` to be interpreted as a type rather than a trait
   |
LL |     let g = BitXor::bitxor;
   |                        +

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0782`.
